app/v1/models/audit/AuditEvent.scala

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.models.audit
+
+case class AuditEvent[T](auditType: String, transactionName: String, detail: T)
```

Comment:
Why is "detail" a generic? If the detail fields (userType, agentReferenceNumber, nino, response, X-CorrelationId) are the same all the time, shouldn't we just make a normal case class?

Reply:
The naming convention I just followed as per our previous Audits, if you have any concerns better ask in the tech channel. And based on the team suggestions we can change

Reply:
The naming convention thing has been resolved by Mark, I'm talking about `detail: T` here, shouldn't this just be a normal case class if the detail field never changes? It always has (userType, agentReferenceNumber, nino, response, X-CorrelationId). Or is this not the case?

Reply:
Yes, For Accounts there is no need to be Generic type... Before we have different audit types. The only reason I didn't change is we are following a template for all the services. So just used the same